packages/replay/src/coreHandlers/performanceObserver.ts

```diff
@@ -1,43 +1,37 @@
-import type { AllPerformanceEntry, ReplayContainer } from '../types';
-import { dedupePerformanceEntries } from '../util/dedupePerformanceEntries';
+import { addPerformanceInstrumentationHandler } from '@sentry-internal/tracing';
+
+import type { ReplayContainer } from '../types';
 
 /**
  * Sets up a PerformanceObserver to listen to all performance entry types.
+ * Returns a callback to stop observing.
  */
-export function setupPerformanceObserver(replay: ReplayContainer): PerformanceObserver {
-  const performanceObserverHandler = (list: PerformanceObserverEntryList): void => {
-    // For whatever reason the observer was returning duplicate navigation
-    // entries (the other entry types were not duplicated).
-    const newPerformanceEntries = dedupePerformanceEntries(
-      replay.performanceEvents,
-      list.getEntries() as AllPerformanceEntry[],
-    );
-    replay.performanceEvents = newPerformanceEntries;
-  };
+export function setupPerformanceObserver(replay: ReplayContainer): () => void {
+  function addPerformanceEntry(entry: PerformanceEntry): void {
+    // It is possible for entries to come up multiple times
+    if (!replay.performanceEvents.includes(entry)) {
+      replay.performanceEvents.push(entry);
+    }
+  }
 
-  const performanceObserver = new PerformanceObserver(performanceObserverHandler);
+  function onEntries({ entries }: { entries: PerformanceEntry[] }): void {
+    entries.forEach(addPerformanceEntry);
+  }
 
-  [
-    'element',
-    'event',
-    'first-input',
-    'largest-contentful-paint',
-    'layout-shift',
-    'longtask',
-    'navigation',
-    'paint',
-    'resource',
-  ].forEach(type => {
-    try {
-      performanceObserver.observe({
-        type,
-        buffered: true,
-      });
-    } catch {
-      // This can throw if an entry type is not supported in the browser.
-      // Ignore these errors.
-    }
+  const clearCallbacks: (() => void)[] = [];
+
+  (['navigation', 'paint', 'resource'] as const).forEach(type => {
+    clearCallbacks.push(addPerformanceInstrumentationHandler(type, onEntries));
   });
 
-  return performanceObserver;
+  clearCallbacks.push(
+    addPerformanceInstrumentationHandler('lcp', ({ metric }) => {
+      replay.lcpMetric = metric;
```

Comment:
Why not push this into performanceEvents?

What would this look like if we wanted to add other metrics (e.g. CLS)?

Reply:
because it is not an performance event, but already the formed metric. Instead of this we could also have a `performanceMetrics` array eventually where we could also push other metrics like CLS in, that would work the same way. Currently we only transform the metric when we send it, but we could eventually refactor this to transform in here and already push the transformed event to a separate array.

Reply:
I changed this a bit to make this more generic - now we keep `performanceEntries` and `replayPerformanceEntries`, which hopefully makes this a bit clearer.